test/e2e/app-dir/next-form/default/next-form.test.ts

```diff
@@ -0,0 +1,223 @@
+              /<Form>'s `.+?` was set to an unsupported value/
+            )
+          )
+        }
+      }
+    )
+  })
+
+  describe('unsupported attributes on submitter', () => {
+    it.each([
+      { name: 'formEncType', baseName: 'encType' },
+      { name: 'formMethod', baseName: 'method' },
+      { name: 'formTarget', baseName: 'target' },
+    ])(
+      'should warn if submitter sets "$name" to an unsupported value and fall back to default submit behavior',
+      async ({ name: attributeName, baseName: attributeBaseName }) => {
+        const session = await next.browser(
+          `/forms/button-formaction-unsupported?attribute=${attributeName}`
+        )
+
+        const submitButton = await session.elementByCss('[type="submit"]')
+        await submitButton.click()
+
+        const logs = await session.log()
+
+        if (isNextDev) {
+          expect(logs).toContainEqual(
+            expect.objectContaining({
+              source: 'error',
+              message: expect.stringContaining(
+                `<Form>'s \`${attributeBaseName}\` was set to an unsupported value`
+              ),
+            })
+          )
+        }
+
+        expect(logs).toContainEqual(
+          expect.objectContaining({
+            source: 'log',
+            message: expect.stringContaining(
+              'correct: default submit behavior was not prevented'
+            ),
+          })
+        )
+        expect(logs).not.toContainEqual(
+          expect.objectContaining({
+            source: 'log',
+            message: expect.stringContaining(
+              'incorrect: default submit behavior was prevented'
+            ),
+          })
+        )
+      }
+    )
+  })
+
+  it('does not push a new history entry if `replace` is passed', async () => {
+    const session = await next.browser(`/forms/with-replace`)
+    const navigationTracker = await trackMpaNavs(session)
+
+    // apparently this is usually not 1...?
+    const prevHistoryLength: number = await session.eval(`history.length`)
+
+    const submitButton = await session.elementByCss('[type="submit"]')
+    await submitButton.click()
+
+    await session.waitForElementByCss('#search-results')
+
+    expect(await navigationTracker.didMpaNavigate()).toBe(false)
+    expect(await session.eval(`history.length`)).toEqual(prevHistoryLength)
+  })
+
+  it('does not navigate if preventDefault is called in onSubmit', async () => {
+    const session = await next.browser(`/forms/with-onsubmit-preventdefault`)
+
+    const submitButton = await session.elementByCss('[type="submit"]')
+    await submitButton.click()
+
+    // see fixture code for explanation why we expect this
+
+    await session.waitForElementByCss('#redirected-results')
+    expect(new URL(await session.url()).pathname).toEqual(
+      '/redirected-from-action'
+    )
+  })
+
+  // TODO(lubieowoce): implement this
+  // (we don't have any methods on BrowserInterface to deal with files)
```

Comment:
might be able to copy something similar to the server action tests which also deal with file upload:

https://github.com/vercel/next.js/blob/c039891908e0eab2e7dcc3d044b12a3fe12abf07/test/e2e/app-dir/actions/app-action.test.ts#L319-L325

Reply:
good one! added a test using this, thanks